### PR TITLE
fix: add validation to jsonResponse

### PR DESCRIPTION
**Pull request by itelo**

### CODE REVIEW

File: packages/nextlove/src/with-route-spec/middlewares/with-validation.ts [hunk 1]
```diff
@@ -114,14 +119,56 @@ const zodIssueToString = (issue: z.ZodIssue) => {
   return `${issue.message} for "${issue.path.join(".")}"`
 }
 
+function validateJsonResponse<JsonResponse extends z.ZodTypeAny>(
+  jsonResponse: JsonResponse | undefined,
+  res: NextApiResponse
+) {
+  let shouldValidateResponsesBasedOnEnv = false
+  if (
+    (process.env.NEXTLOVE_VALIDATION_RESPONSES === undefined &&
```
Comment (seveibar) [inline packages/nextlove/src/with-route-spec/middlewares/with-validation.ts:128]: IIRC empty environment variables can be `""`

```suggestion
    (!Boolean(process.env.NEXTLOVE_VALIDATION_RESPONSES) &&
```

File: packages/nextlove/src/with-route-spec/middlewares/with-validation.ts [hunk 2]
```diff
@@ -114,14 +119,56 @@ const zodIssueToString = (issue: z.ZodIssue) => {
...
+      process.env.NODE_ENV !== "production") ||
+    process.env.NEXTLOVE_VALIDATION_RESPONSES === "true"
```
Comment (seveibar) [inline packages/nextlove/src/with-route-spec/middlewares/with-validation.ts:130]:
```suggestion
    process.env.NEXTLOVE_VALIDATE_RESPONSES === "true"
```

File: packages/nextlove/src/with-route-spec/middlewares/with-validation.ts [hunk 3]
```diff
@@ -114,14 +119,56 @@ const zodIssueToString = (issue: z.ZodIssue) => {
   return `${issue.message} for "${issue.path.join(".")}"`
 }
 
+function validateJsonResponse<JsonResponse extends z.ZodTypeAny>(
```
Comment (seveibar) [inline packages/nextlove/src/with-route-spec/middlewares/with-validation.ts:122]: Function should do as it's named

```suggestion
function conditionallyValidateJsonResponse<JsonResponse extends z.ZodTypeAny>(
```

a better alternative to this is to have `shouldValidateResponses` as a separate function, then in the middleware do

```ts
if (shouldValidateResponses()) {
  validateJsonResponse(...)
}
```

**Review by seveibar — COMMENTED** [review]
minor naming but looks good

**Review by codetheweb — COMMENTED** [review]
maybe it'd be better to have this as a new option on `createWithRouteSpec()` so consumers can choose how to conditionally enable & disable it

**Review by seveibar — APPROVED** [review]
Ideally we used status, not the presence of an error, to determine if we should do validation. We can create a new issue to do that or solve here.

**Pull request merged**